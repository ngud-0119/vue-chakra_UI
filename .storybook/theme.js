import { create } from '@storybook/theming';

export default create({
  base: 'light',
  brandTitle: 'chakra-ui/vue storybook',
  brandUrl: 'https://github.com/chakra-ui/chakra-ui-vue',
  brandImage: 'https://res.cloudinary.com/xtellar/image/upload/v1584242872/chakra-ui/chakra-ui-vue-beta.png'
});
