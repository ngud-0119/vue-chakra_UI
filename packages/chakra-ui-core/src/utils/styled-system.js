
/**
 * Existential getter function that can be used in any style declaration to get a value
 * from theme, with support for fallback values. This helps prevent errors from
 * throwing when a theme value is missing.
 * @param {Object} obj Theme property
 * @param {String} key Theme key
 * @param {String} def Definition if non-existent
 * @param {*} p
 * @param {*} undef
 */
export const __get = (obj, key, def, p, undef) => {
  key = key && key.split ? key.split('.') : [key]

  for (p = 0; p < key.length; p++) {
    obj = obj ? obj[key[p]] : undef
  }

  return obj === undef ? def : obj
}
