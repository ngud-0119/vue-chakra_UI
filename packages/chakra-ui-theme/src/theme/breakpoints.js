/**
 * @description These breakpoint styles were adapted from [@chakra-ui](https://chakra-ui.com/)
 */

const _breakpoints = {
  sm: '30em',
  md: '48em',
  lg: '62em',
  xl: '80em',
  '2xl': '96em'
}

export const createBreakpoints = (
  config = {}
) => {
  return { base: '0em', ...config }
}

const breakpoints = createBreakpoints(_breakpoints)

export default breakpoints
