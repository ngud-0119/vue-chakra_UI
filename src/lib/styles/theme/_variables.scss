/* THEME COLORS */
$primary: hsl(209, 100%, 50%);
$secondary: hsl(278, 100%, 69%);
$success: hsl(160, 97%, 42%);
$warning: hsl(40, 100%, 50%);
$danger: hsl(350, 100%, 56%);
$light: hsl(0, 0%, 90%);
$dark: hsl(221, 15%, 29%);