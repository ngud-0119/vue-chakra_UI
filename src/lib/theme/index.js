import colors from './colors'
import typography from './typography'
import borders from './borders'
import opacity from './opacity'
import radii from './radii'
import shadows from './shadows'
import sizes, { baseSizes } from './sizes'
import zIndices from './z-indices'

const space = baseSizes

const theme = {
  colors,
  borders,
  opacity,
  radii,
  shadows,
  sizes,
  space,
  zIndices,
  ...typography
}

export default theme
