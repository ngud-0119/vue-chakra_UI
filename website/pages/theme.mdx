import SEO from '../components/SEO'

<SEO
  title="Theme"
  description="The theme object is where you define your application's color palette, type scale, font stacks, breakpoints, border radius values, and more"
/>


import {
  ColorPalette,
  ColorPalettes,
  ColorWrapper,
} from "../components/ColorPalette";

# Theme

The theme object is where you define your application's color palette, type
scale, font stacks, breakpoints, border radius values, and more. Theming with
Chakra UI is based on the
[Styled System Theme Specification](https://system-ui.com/theme/)

<carbon-ad />

## Colors

Add a `theme.colors` object to provide colors for your project. By default these
colors can be referenced by the `color`, `borderColor`, and `backgroundColor`,
`fill`, `stroke`, styles.

We recommend adding palette that go from `50` - `900`. Use tools like
[Coolors](https://coolors.co/app) or [Palx](https://palx.jxnblk.com) to generate
these palette.


```js
// theme.js
export default {
  colors: {
    transparent: "transparent",
    black: "#000",
    white: "#fff",
    gray: {
      50: "#f7fafc",
      // ...
      900: "#1a202c",
    },
    // ...
  },
};
```

By default, Chakra UI exposes a [**default theme object**](https://github.com/chakra-ui/chakra-ui-vue/blob/master/packages/chakra-ui-theme/src/theme/index.js) that follows the [System UI Theme Specification](https://system-ui.com/theme/).

<!-- Apply Color Palettes here -->
### Black & White

<ColorWrapper>
  <ColorPalette color="black" name="Black" />
  <ColorPalette color="white" name="White" />
  <div />
</ColorWrapper>

### Gray

<ColorWrapper>
  <ColorPalettes color="gray" />
</ColorWrapper>

### Cyan

<ColorWrapper>
  <ColorPalettes color="cyan" />
</ColorWrapper>

### Vue

<ColorWrapper>
  <ColorPalettes color="vue" />
</ColorWrapper>

### Green

<ColorWrapper>
  <ColorPalettes color="green" />
</ColorWrapper>

### Orange

<ColorWrapper>
  <ColorPalettes color="orange" />
</ColorWrapper>

### Red

<ColorWrapper>
  <ColorPalettes color="red" />
</ColorWrapper>

### Yellow

<ColorWrapper>
  <ColorPalettes color="yellow" />
</ColorWrapper>

### Indigo

<ColorWrapper>
  <ColorPalettes color="indigo" />
</ColorWrapper>

### Pink

<ColorWrapper>
  <ColorPalettes color="pink" />
</ColorWrapper>

### Blue

<ColorWrapper>
  <ColorPalettes color="blue" />
</ColorWrapper>


## Breakpoints

To configure the default breakpoints used in responsive array values, add a
breakpoints array to your theme. These values will be used to generate
mobile-first (i.e. min-width) media queries, which can then be used to apply
responsive styles.

> For example, you can write `<c-box :font-size="['14px', '16px']"/>` to make the
> fontSize responsive.

```js
// theme.js
export default {
  breakpoints: ["30em", "48em", "62em", "80em"],
};
```

## Spacing

The `space` key allows you to customize the global spacing and sizing scale for
your project. By default these spacing value can be referenced by the `padding`,
`margin`, and `top`, `left`, `right`, `bottom` styles.

```js
export default {
  space: {
    px: "1px",
    "0": "0",
    "1": "0.25rem",
    "2": "0.5rem",
    "3": "0.75rem",
    "4": "1rem",
    "5": "1.25rem",
    "6": "1.5rem",
    "8": "2rem",
    "10": "2.5rem",
    "12": "3rem",
    "16": "4rem",
    "20": "5rem",
    "24": "6rem",
    "32": "8rem",
    "40": "10rem",
    "48": "12rem",
    "56": "14rem",
    "64": "16rem",
  },
};
```

By default, Chakra includes a comprehensive numeric spacing scale inspired by
Tailwind CSS. The values are proportional, so 1 spacing unit is equal to
`0.25rem`, which translates to `4px` by default in common browsers.

> **Mental model:** If you need a spacing of `40px`, divide it by `4`. That'll
> give you `10`. Then use it in your component.

| Name | Space   | Pixels |                                     |
| ---- | ------- | ------ | ----------------------------------  |
| 0    | 0       | 0px    | <c-box bg="pink.200" h="4" w="0"/>  |
| px   | 1px     | 1px    | <c-box bg="pink.200" h="4" w="px"/> |
| 1    | 0.25rem | 4px    | <c-box bg="pink.200" h="4" w="1"/>  |
| 2    | 0.5rem  | 8px    | <c-box bg="pink.200" h="4" w="2"/>  |
| 3    | 0.75rem | 12px   | <c-box bg="pink.200" h="4" w="3"/>  |
| 4    | 1rem    | 16px   | <c-box bg="pink.200" h="4" w="4"/>  |
| 5    | 1.25rem | 20px   | <c-box bg="pink.200" h="4" w="5"/>  |
| 6    | 1.5rem  | 24px   | <c-box bg="pink.200" h="4" w="6"/>  |
| 8    | 2rem    | 32px   | <c-box bg="pink.200" h="4" w="8"/>  |
| 10   | 2.5rem  | 40px   | <c-box bg="pink.200" h="4" w="10"/> |
| 12   | 3rem    | 48px   | <c-box bg="pink.200" h="4" w="12"/> |
| 16   | 4rem    | 64px   | <c-box bg="pink.200" h="4" w="16"/> |
| 20   | 5rem    | 80px   | <c-box bg="pink.200" h="4" w="20"/> |
| 24   | 6rem    | 96px   | <c-box bg="pink.200" h="4" w="24"/> |
| 32   | 8rem    | 128px  | <c-box bg="pink.200" h="4" w="32"/> |
| 40   | 10rem   | 160px  | <c-box bg="pink.200" h="4" w="40"/> |
| 48   | 12rem   | 192px  | <c-box bg="pink.200" h="4" w="48"/> |
| 56   | 14rem   | 224px  | <c-box bg="pink.200" h="4" w="56"/> |
| 64   | 16rem   | 256px  | <c-box bg="pink.200" h="4" w="64"/> |

## Sizes

The `sizes` key allows you to customize the global sizing of components you
build for your project. By default these spacing value can be referenced by the
`width`, `height`, and `maxWidth`, `minWidth`, `maxHeight`, `minHeight` styles.

```js
export default {
  sizes: {
    ...theme.space,
    full: "100%",
    "3xs": "14rem",
    "2xs": "16rem",
    xs: "20rem",
    sm: "24rem",
    md: "28rem",
    lg: "32rem",
    xl: "36rem",
    "2xl": "42rem",
    "3xl": "48rem",
    "4xl": "56rem",
    "5xl": "64rem",
    "6xl": "72rem",
  },
};
```

For a component like this: `<c-box w="4" h="3" />` will generate an empty `div`
with width set to `1rem` or `16px` and height set to `0.75rem` or `12px`

## Z-Index

Chakra provides some zIndex values out of the box to control the stacking order
of components.

```js
export default {
  zIndices: {
    hide: -1,
    auto: "auto",
    base: 0,
    docked: 10,
    dropdown: 1000,
    sticky: 1100,
    banner: 1200,
    overlay: 1300,
    modal: 1400,
    popover: 1500,
    skipLink: 1600,
    toast: 1700,
    tooltip: 1800,
  },
};
```