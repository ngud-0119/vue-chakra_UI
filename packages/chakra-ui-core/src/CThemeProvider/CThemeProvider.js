/**
 * Hey! Welcome to @chakra-ui/vue CThemeProvider
 *
 * The CThemeProvider componeent provides theme context to all it's
 * children.
 *
 * @see Docs     https://vue.chakra-ui.com/getting-started
 * @see Source   https://github.com/chakra-ui/chakra-ui-vue/blob/master/packages/chakra-ui-core/src/CThemeProvider/CThemeProvider.js
 */

import { injectGlobal } from '@emotion/css'
import { colorModeObserver } from '../utils/color-mode-observer'

/**
 * CThemeProvider component
 */
const CThemeProvider = {
  name: 'CThemeProvider',
  provide () {
    return {
      $chakraTheme: () => this.theme,
      $chakraIcons: this.icons,
      /**
       * By default the ThemeProvider exposes a colorMode value of light
       * If no `ColorModeProvider` is provided in children/ consumer app, all chakra
       * components will consume the $chakraColorMode from here.
       */
      $chakraColorMode: () => 'light'
    }
  },
  props: {
    rootKey: {
      type: String,
      default: ':root'
    }
  },
  computed: {
    icons () {
      return this.$chakra ? this.$chakra.icons : {}
    },
    theme () {
      return this.$chakra.theme
    }
  },
  watch: {
    theme: {
      immediate: true,
      handler (newVal) {
        colorModeObserver.theme = newVal
        this.updateGlobalCssVars()
      }
    },
    icons: {
      immediate: true,
      handler (newVal) {
        colorModeObserver.icons = newVal
      }
    }
  },
  methods: {
    updateGlobalCssVars () {
      const rootKey = this.rootKey || ':root'
      injectGlobal({
        [rootKey]: this.theme.__cssVars
      })
    }
  },
  render (h) {
    return h('div', {
      attrs: {
        id: '__chakra-app',
        'data-chakra-component': 'CThemeProvider'
      }
    }, this.$slots.default)
  }
}

export default CThemeProvider
