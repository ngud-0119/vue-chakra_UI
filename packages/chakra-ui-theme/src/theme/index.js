import colors from './colors'
import typography from './typography'
import borders, { borderWidths } from './borders'
import opacity from './opacity'
import radii from './radii'
import shadows from './shadows'
import sizes, { baseSizes } from './sizes'
import zIndices from './z-indices'
import breakpoints from './breakpoints'

const space = baseSizes

const theme = {
  breakpoints,
  zIndices,
  radii,
  opacity,
  borders,
  colors,
  ...typography,
  borderWidths,
  sizes,
  shadows,
  space
}

export default theme
