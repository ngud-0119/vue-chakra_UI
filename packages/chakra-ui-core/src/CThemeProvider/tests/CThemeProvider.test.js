import { toCSSVar } from '@chakra-ui/styled-system'
import { mount, shallowMount } from '@vue/test-utils'
import theme from '@chakra-ui/theme-vue'
import CThemeProvider from '..'
import { getElementStyles } from '@/tests/test-utils'

describe('CThemeProvider', () => {
  const ChildComponent = {
    inject: ['$chakraTheme', '$chakraColorMode'],
    render: h => h('div', {})
  }

  it('should be a Vue component', () => {
    const themeProvider = shallowMount(CThemeProvider, {
      mocks: {
        $chakra: {
          theme
        }
      },
      slots: {
        default: [ChildComponent]
      }
    })
    expect(themeProvider.isVueInstance()).toBeTruthy()
  })

  it('should provide theme & default color mode to child components', () => {
    const themeProvider = shallowMount(CThemeProvider, {
      mocks: {
        $chakra: {
          theme
        }
      },
      slots: {
        default: [ChildComponent]
      }
    })
    expect(themeProvider.find(ChildComponent).vm.$chakraTheme()).toBe(theme)
    expect(themeProvider.find(ChildComponent).vm.$chakraColorMode()).toBe('light')
  })

  it('should inject global Css variables to root', () => {
    const rootKey = ':root'

    mount(CThemeProvider, {
      props: {
        rootKey
      },
      mocks: {
        $chakra: {
          theme: toCSSVar(theme)
        }
      },
      slots: {
        default: [ChildComponent]
      }
    })

    const root = document.querySelector(rootKey)
    expect(getElementStyles(root)).toMatchSnapshot()
  })
})
