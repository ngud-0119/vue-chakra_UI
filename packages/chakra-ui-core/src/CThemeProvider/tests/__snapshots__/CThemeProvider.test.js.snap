// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CThemeProvider should inject global Css variables to root 1`] = `
"--chakra-ring-inset: var(--chakra-empty, );
 --chakra-ring-offset-width: 0px;
 --chakra-ring-offset-color: #fff;
 --chakra-ring-color: rgba(66, 153, 225, 0.6);
 --chakra-ring-offset-shadow: 0 0 #0000;
 --chakra-ring-shadow: 0 0 #0000;
 --chakra-space-x-reverse: 0;
 --chakra-space-y-reverse: 0;
 --chakra-colors-transparent: transparent;
 --chakra-colors-current: currentColor;
 --chakra-colors-black: #000;
 --chakra-colors-white: #fff;
 --chakra-colors-primary: #1a72ff;
 --chakra-colors-secondary: #7425f3;
 --chakra-colors-tertiary: #fe9e4d;
 --chakra-colors-success: #3ea76a;
 --chakra-colors-warning: #ffc61a;
 --chakra-colors-error: #dd3b4b;
 --chakra-colors-whiteAlpha-50: rgba(255, 255, 255, 0.04);
 --chakra-colors-whiteAlpha-100: rgba(255, 255, 255, 0.06);
 --chakra-colors-whiteAlpha-200: rgba(255, 255, 255, 0.08);
 --chakra-colors-whiteAlpha-300: rgba(255, 255, 255, 0.16);
 --chakra-colors-whiteAlpha-400: rgba(255, 255, 255, 0.24);
 --chakra-colors-whiteAlpha-500: rgba(255, 255, 255, 0.36);
 --chakra-colors-whiteAlpha-600: rgba(255, 255, 255, 0.48);
 --chakra-colors-whiteAlpha-700: rgba(255, 255, 255, 0.64);
 --chakra-colors-whiteAlpha-800: rgba(255, 255, 255, 0.80);
 --chakra-colors-whiteAlpha-900: rgba(255, 255, 255, 0.92);
 --chakra-colors-blackAlpha-50: rgba(0, 0, 0, 0.04);
 --chakra-colors-blackAlpha-100: rgba(0, 0, 0, 0.06);
 --chakra-colors-blackAlpha-200: rgba(0, 0, 0, 0.08);
 --chakra-colors-blackAlpha-300: rgba(0, 0, 0, 0.16);
 --chakra-colors-blackAlpha-400: rgba(0, 0, 0, 0.24);
 --chakra-colors-blackAlpha-500: rgba(0, 0, 0, 0.36);
 --chakra-colors-blackAlpha-600: rgba(0, 0, 0, 0.48);
 --chakra-colors-blackAlpha-700: rgba(0, 0, 0, 0.64);
 --chakra-colors-blackAlpha-800: rgba(0, 0, 0, 0.80);
 --chakra-colors-blackAlpha-900: rgba(0, 0, 0, 0.92);
 --chakra-colors-gray-50: #F7FAFC;
 --chakra-colors-gray-100: #EDF2F7;
 --chakra-colors-gray-200: #E2E8F0;
 --chakra-colors-gray-300: #CBD5E0;
 --chakra-colors-gray-400: #A0AEC0;
 --chakra-colors-gray-500: #718096;
 --chakra-colors-gray-600: #4A5568;
 --chakra-colors-gray-700: #2D3748;
 --chakra-colors-gray-800: #1A202C;
 --chakra-colors-gray-900: #171923;
 --chakra-colors-cyan-50: #EDFDFD;
 --chakra-colors-cyan-100: #C4F1F9;
 --chakra-colors-cyan-200: #9DECF9;
 --chakra-colors-cyan-300: #76E4F7;
 --chakra-colors-cyan-400: #0BC5EA;
 --chakra-colors-cyan-500: #00B5D8;
 --chakra-colors-cyan-600: #00A3C4;
 --chakra-colors-cyan-700: #0987A0;
 --chakra-colors-cyan-800: #086F83;
 --chakra-colors-cyan-900: #065666;
 --chakra-colors-vue-50: #e1fbf0;
 --chakra-colors-vue-100: #c0ecda;
 --chakra-colors-vue-200: #9edfc2;
 --chakra-colors-vue-300: #79d0aa;
 --chakra-colors-vue-400: #55c392;
 --chakra-colors-vue-500: #3caa79;
 --chakra-colors-vue-600: #2d845e;
 --chakra-colors-vue-700: #1d5e42;
 --chakra-colors-vue-800: #0c3a26;
 --chakra-colors-vue-900: #001508;
 --chakra-colors-green-50: #e2fbed;
 --chakra-colors-green-100: #c2ebd4;
 --chakra-colors-green-200: #9fddb9;
 --chakra-colors-green-300: #7ccf9e;
 --chakra-colors-green-400: #58c184;
 --chakra-colors-green-500: #3ea76a;
 --chakra-colors-green-600: #2e8251;
 --chakra-colors-green-700: #1f5d3a;
 --chakra-colors-green-800: #0f3921;
 --chakra-colors-green-900: #001506;
 --chakra-colors-orange-50: #ffefdc;
 --chakra-colors-orange-100: #ffd4ae;
 --chakra-colors-orange-200: #ffba7f;
 --chakra-colors-orange-300: #fe9e4d;
 --chakra-colors-orange-400: #fd821b;
 --chakra-colors-orange-500: #e46902;
 --chakra-colors-orange-600: #b25100;
 --chakra-colors-orange-700: #7f3900;
 --chakra-colors-orange-800: #4e2200;
 --chakra-colors-orange-900: #1f0900;
 --chakra-colors-red-50: #ffe5e9;
 --chakra-colors-red-100: #f9bcc2;
 --chakra-colors-red-200: #ee919a;
 --chakra-colors-red-300: #e66673;
 --chakra-colors-red-400: #dd3b4b;
 --chakra-colors-red-500: #c42231;
 --chakra-colors-red-600: #991826;
 --chakra-colors-red-700: #6e101a;
 --chakra-colors-red-800: #44070e;
 --chakra-colors-red-900: #1e0001;
 --chakra-colors-yellow-50: #fff9da;
 --chakra-colors-yellow-100: #ffecad;
 --chakra-colors-yellow-200: #ffdf7d;
 --chakra-colors-yellow-300: #ffd24b;
 --chakra-colors-yellow-400: #ffc61a;
 --chakra-colors-yellow-500: #e6ac00;
 --chakra-colors-yellow-600: #b38600;
 --chakra-colors-yellow-700: #806000;
 --chakra-colors-yellow-800: #4e3900;
 --chakra-colors-yellow-900: #1d1300;
 --chakra-colors-indigo-50: #f2e5ff;
 --chakra-colors-indigo-100: #d2b5ff;
 --chakra-colors-indigo-200: #b285fa;
 --chakra-colors-indigo-300: #9356f7;
 --chakra-colors-indigo-400: #7425f3;
 --chakra-colors-indigo-500: #5a0cda;
 --chakra-colors-indigo-600: #4608aa;
 --chakra-colors-indigo-700: #32057b;
 --chakra-colors-indigo-800: #1e024c;
 --chakra-colors-indigo-900: #0c001e;
 --chakra-colors-pink-50: #fff5f7;
 --chakra-colors-pink-100: #fed7e2;
 --chakra-colors-pink-200: #fbb6ce;
 --chakra-colors-pink-300: #f687b3;
 --chakra-colors-pink-400: #ed64a6;
 --chakra-colors-pink-500: #d53f8c;
 --chakra-colors-pink-600: #b83280;
 --chakra-colors-pink-700: #97266d;
 --chakra-colors-pink-800: #702459;
 --chakra-colors-pink-900: #521B41;
 --chakra-colors-blue-50: #def0ff;
 --chakra-colors-blue-100: #afd0ff;
 --chakra-colors-blue-200: #7db1ff;
 --chakra-colors-blue-300: #4b91ff;
 --chakra-colors-blue-400: #1a72ff;
 --chakra-colors-blue-500: #0058e6;
 --chakra-colors-blue-600: #0045b4;
 --chakra-colors-blue-700: #003182;
 --chakra-colors-blue-800: #001d51;
 --chakra-colors-blue-900: #000a21;
 --chakra-borders-none: 0;
 --chakra-borders-1px: 1px solid;
 --chakra-borders-2px: 2px solid;
 --chakra-borders-4px: 4px solid;
 --chakra-borderWidths-sm: 1px;
 --chakra-borderWidths-md: 2px;
 --chakra-borderWidths-lg: 4px;
 --chakra-fonts-heading: -apple-system,BlinkMacSystemFont,\\"Segoe UI\\",Helvetica,Arial,sans-serif,\\"Apple Color Emoji\\",\\"Segoe UI Emoji\\",\\"Segoe UI Symbol\\";
 --chakra-fonts-body: -apple-system,BlinkMacSystemFont,\\"Segoe UI\\",Helvetica,Arial,sans-serif,\\"Apple Color Emoji\\",\\"Segoe UI Emoji\\",\\"Segoe UI Symbol\\";
 --chakra-fonts-mono: SFMono-Regular,Menlo,Monaco,Consolas,\\"Liberation Mono\\",\\"Courier New\\",monospace;
 --chakra-fontSizes-xs: 0.75rem;
 --chakra-fontSizes-sm: 0.875rem;
 --chakra-fontSizes-md: 1rem;
 --chakra-fontSizes-lg: 1.125rem;
 --chakra-fontSizes-xl: 1.25rem;
 --chakra-fontSizes-2xl: 1.5rem;
 --chakra-fontSizes-3xl: 1.875rem;
 --chakra-fontSizes-4xl: 2.25rem;
 --chakra-fontSizes-5xl: 3rem;
 --chakra-fontSizes-6xl: 4rem;
 --chakra-fontWeights-hairline: 100;
 --chakra-fontWeights-thin: 200;
 --chakra-fontWeights-light: 300;
 --chakra-fontWeights-normal: 400;
 --chakra-fontWeights-medium: 500;
 --chakra-fontWeights-semibold: 600;
 --chakra-fontWeights-bold: 700;
 --chakra-fontWeights-extrabold: 800;
 --chakra-fontWeights-black: 900;
 --chakra-letterSpacings-tighter: -0.05em;
 --chakra-letterSpacings-tight: -0.025em;
 --chakra-letterSpacings-normal: 0;
 --chakra-letterSpacings-wide: 0.025em;
 --chakra-letterSpacings-wider: 0.05em;
 --chakra-letterSpacings-widest: 0.1em;
 --chakra-lineHeights-normal: normal;
 --chakra-lineHeights-none: 1;
 --chakra-lineHeights-shorter: 1.25;
 --chakra-lineHeights-short: 1.375;
 --chakra-lineHeights-base: 1.5;
 --chakra-lineHeights-tall: 1.625;
 --chakra-lineHeights-taller: 2;
 --chakra-radii-none: 0;
 --chakra-radii-sm: 0.125rem;
 --chakra-radii-md: 0.25rem;
 --chakra-radii-lg: 0.5rem;
 --chakra-radii-full: 9999px;
 --chakra-space-0: 0;
 --chakra-space-1: 0.25rem;
 --chakra-space-2: 0.5rem;
 --chakra-space-3: 0.75rem;
 --chakra-space-4: 1rem;
 --chakra-space-5: 1.25rem;
 --chakra-space-6: 1.5rem;
 --chakra-space-8: 2rem;
 --chakra-space-10: 2.5rem;
 --chakra-space-12: 3rem;
 --chakra-space-16: 4rem;
 --chakra-space-20: 5rem;
 --chakra-space-24: 6rem;
 --chakra-space-32: 8rem;
 --chakra-space-40: 10rem;
 --chakra-space-48: 12rem;
 --chakra-space-56: 14rem;
 --chakra-space-64: 16rem;
 --chakra-space-px: 1px;
 --chakra-shadows-sm: 0 1px 3px 0 rgba(0, 0, 0, 0.1),0 1px 2px 0 rgba(0, 0, 0, 0.06);
 --chakra-shadows-md: 0 4px 6px -1px rgba(0, 0, 0, 0.1),0 2px 4px -1px rgba(0, 0, 0, 0.06);
 --chakra-shadows-lg: 0 10px 15px -3px rgba(0, 0, 0, 0.1),0 4px 6px -2px rgba(0, 0, 0, 0.05);
 --chakra-shadows-xl: 0 20px 25px -5px rgba(0, 0, 0, 0.1),0 10px 10px -5px rgba(0, 0, 0, 0.04);
 --chakra-shadows-2xl: 0 25px 50px -12px rgba(0, 0, 0, 0.25);
 --chakra-shadows-outline: 0 0 0 3px rgba(66, 153, 225, 0.6);
 --chakra-shadows-inner: inset 0 2px 4px 0 rgba( 0, 0, 0, 0.06);
 --chakra-shadows-none: none;
 --chakra-sizes-0: 0;
 --chakra-sizes-1: 0.25rem;
 --chakra-sizes-2: 0.5rem;
 --chakra-sizes-3: 0.75rem;
 --chakra-sizes-4: 1rem;
 --chakra-sizes-5: 1.25rem;
 --chakra-sizes-6: 1.5rem;
 --chakra-sizes-8: 2rem;
 --chakra-sizes-10: 2.5rem;
 --chakra-sizes-12: 3rem;
 --chakra-sizes-16: 4rem;
 --chakra-sizes-20: 5rem;
 --chakra-sizes-24: 6rem;
 --chakra-sizes-32: 8rem;
 --chakra-sizes-40: 10rem;
 --chakra-sizes-48: 12rem;
 --chakra-sizes-56: 14rem;
 --chakra-sizes-64: 16rem;
 --chakra-sizes-px: 1px;
 --chakra-sizes-full: 100%;
 --chakra-sizes-3xs: 14rem;
 --chakra-sizes-2xs: 16rem;
 --chakra-sizes-xs: 20rem;
 --chakra-sizes-sm: 24rem;
 --chakra-sizes-md: 28rem;
 --chakra-sizes-lg: 32rem;
 --chakra-sizes-xl: 36rem;
 --chakra-sizes-2xl: 42rem;
 --chakra-sizes-3xl: 48rem;
 --chakra-sizes-4xl: 56rem;
 --chakra-sizes-5xl: 64rem;
 --chakra-sizes-6xl: 72rem;
 --chakra-sizes-containers-sm: 640px;
 --chakra-sizes-containers-md: 768px;
 --chakra-sizes-containers-lg: 1024px;
 --chakra-sizes-containers-xl: 1280px;
 --chakra-zIndices-hide: -1;
 --chakra-zIndices-auto: auto;
 --chakra-zIndices-base: 0;
 --chakra-zIndices-docked: 10;
 --chakra-zIndices-dropdown: 1000;
 --chakra-zIndices-sticky: 1100;
 --chakra-zIndices-banner: 1200;
 --chakra-zIndices-overlay: 1300;
 --chakra-zIndices-modal: 1400;
 --chakra-zIndices-popover: 1500;
 --chakra-zIndices-skipLink: 1600;
 --chakra-zIndices-toast: 1700;
 --chakra-zIndices-tooltip: 1800;
"
`;
